Classes/Controller/AbstractController.php

```diff
@@ -294,7 +298,7 @@ protected function redirectByAction($action = 'new', $category = 'redirect')
 
         // if redirect target
         if ($target) {
-            $this->uriBuilder->setTargetPageUid($target);
+            $this->uriBuilder->setTargetPageUid((int) $target);
```

Comment:
@Baachi 
Just a hint, or comment, which has nothing to do with your int casting, but more general.

I am just testing your changes in TYPO3 10 and stumbled upon your fork and this this issue:

As the typoscript returns a link already, `$target` is a string and there is no need for using the uriBuilder.

	plugin.tx_femanager {
		settings {
			edit {
				# redirect user (or admin if adminconfirmation) after profile creation (could be a PID, URL or fileadmin-Link)
				redirect = TEXT
				redirect {
					typolink {
						parameter = 1
						returnLast = url
					}
				}
			}
		}
	}

So I am not sure, if there should be a check like this:

    if ($target) {
		if ( is_int($target) || is_numeric($target) ) {
			$this->uriBuilder->setTargetPageUid((int) $target);
			$this->uriBuilder->setLinkAccessRestrictedPages(true);
			$link = $this->uriBuilder->build();
			$this->redirectToUri(StringUtility::removeDoubleSlashesFromUri($link));
		} else {
			$this->redirectToUri(StringUtility::removeDoubleSlashesFromUri($target));
		}
	}

Reply:
This fix worked for me.